docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,567 @@
+          providerRef:
+            name: aws-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.allocatedStorage"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+</div>
+<div class="tab-pane fade" id="gcp-tab-1" markdown="1">
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.gcp.database.example.org
+  labels:
+    provider: gcp
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: database.gcp.crossplane.io/v1beta1
+        kind: CloudSQLInstance
+        spec:
+          forProvider:
+            databaseVersion: MYSQL_5_7
+            region: us-central1
+            ipConfiguration:
+              ipv4Enabled: true
+              authorizedNetworks:
+                - value: "0.0.0.0/0"
+            settings:
+              tier: db-n1-standard-1
+              dataDiskType: PD_SSD
+              dataDiskSizeGb: 10
+          writeConnectionSecretToRef:
+            namespace: crossplane-system
+          providerRef:
+            name: gcp-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.settings.dataDiskSizeGb"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+</div>
+<div class="tab-pane fade" id="azure-tab-1" markdown="1">
+> Note: the `Composition` for Azure also includes a `ResourceGroup` and
+> `MySQLServerFirewallRule` that are required to provision a publicly available
+> MySQL instance on Azure. Composition enables scenarios such as this, as well
+> as far more complex ones. See the [composition] documentation for more
+> information.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.azure.database.example.org
+  labels:
+    provider: azure
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: azure.crossplane.io/v1alpha3
+        kind: ResourceGroup
+        spec:
```

Comment:
`spec.location` is a required field of `ResourceGroup`. We don't seem to have this validation in CRD.
```
'create failed: cannot create ResourceGroup: resources.GroupsClient#CreateOrUpdate:
      Invalid input: autorest/validation: validation failed: parameter=parameters.Location
      constraint=Null value=(*string)(nil) details: value can not be null; required
      parameter'
```